### DISCUSSION TITLE
RearchConsumer vs StatelessWidget

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#186] [Q&A] [answered]
If I have a `RearchConsumer` where I'm not using for the moment the `WidgetHandle`... Must I transform it to `StatelessWidget` for performance reasons or it isn't expensive so I can continue to declare it as `RearchConsumer`?

Thanks

**Comment by GregoryConrad** [comment, marked as answer]
You won’t notice a difference. StatelessWidget would be the proper solution here since you don’t need the handle, but either should be fine performance wise, especially since you don’t use the handle.

the @rearchWidget macro actually already switches between the two automatically for you, but obviously macros are still half-baked at the moment and aren’t stable yet.

**Reply by busslina** [reply]
Cool !!
